app/models/concerns/foreman_azure_rm/vm_extensions/managed_vm.rb

```diff
@@ -88,6 +88,8 @@ def create_nics(region, args = {})
                                   NetworkModels::IPAllocationMethod::Dynamic
                                 when 'None'
                                   nil
+                                else
+                                    raise RuntimeError, "Invalid Public IP type"
```

Comment:
Can you specify valid options here? It's annoying to guess what values are accepted.

Reply:
done.